lib/driver.js

```diff
@@ -172,18 +172,12 @@ class XCUITestDriver extends BaseDriver {
 
     if (this.opts.browserName) {
       log.info('Safari test requested');
-      this.safari = true;      
+      this.safari = true;
       this.opts.app = undefined;
       this.opts.processArguments = this.opts.processArguments || {};
-      if (this.isRealDevice()) {
-        this.opts.bundleId = SAFARI_LAUNCHER_BUNDLE;
-        this._currentUrl = this.opts.safariInitialUrl || `http://appium.io`;
-        this.opts.processArguments.args = [this._currentUrl];
-      } else {
-        this.opts.bundleId = SAFARI_BUNDLE_ID;
-        this._currentUrl = this.opts.safariInitialUrl || `http://${this.opts.address}:${this.opts.port}/welcome`;
-        this.opts.processArguments.args = ['-u', this._currentUrl];
-      }
+      this.opts.bundleId = SAFARI_BUNDLE_ID;
+      this._currentUrl = this.opts.safariInitialUrl || DEFAULT_SAFARI_URL;
```

Comment:
This needs to be different for simulators as real devices. It's the only part of the conditional that remains.

```js
this._currentUrl = this.opts.safariInitialUrl || 
                   this.isRealDevice() ? 'http://appium.io' : 
                   `http://${this.opts.address}:${this.opts.port}/welcome`;
```